iio/iio_adxrs290/iio_adxrs290.c

```diff
@@ -178,3 +179,61 @@ ssize_t set_adxrs290_iio_ch_lpf(void *device, char *buf, size_t len,
 
 	return FAILURE;
 }
+
+int32_t adxrs290_update_active_channels(void *device, uint32_t mask)
+{
+	struct adxrs290_dev *dev = device;
+
+	adxrs290_ch_mask = mask;
+
+	return SUCCESS;
+}
+
+int32_t adxrs290_close_channels(void *device)
+{
+	struct adxrs290_dev *dev = device;
+
+	adxrs290_ch_mask = 0;
+
+	return SUCCESS;
+}
+
+static bool get_next_ch_idx(uint32_t ch_mask, uint32_t last_idx,
+			    uint32_t *new_idx)
+{
+	last_idx++;
+	ch_mask >>= last_idx;
+	if (!ch_mask) {
+		*new_idx = -1;
+		return 0;
+	}
+	while (!(ch_mask & 1)) {
+		last_idx++;
+		ch_mask >>= 1;
+	}
+	*new_idx = last_idx;
+
+	return 1;
+}
+
+int32_t	adxrs290_read_samples(void *device, uint16_t *buff,
+			      uint32_t nb_samples)
+{
+	struct adxrs290_dev	*dev = device;
+	uint32_t		ch_idx;
+	uint32_t		i;
+	uint32_t		k;
+	int16_t			data;
+
+	ch_idx = -1;
+	k = 0;
+	for (i = 0; i < nb_samples; i++)
+		while (get_next_ch_idx(adxrs290_ch_mask, ch_idx, &ch_idx)) {
+			adxrs290_get_rate_data(dev, ch_idx, &data);
+			if (ch_idx == ADXRS290_CHANNEL_TEMP)
+				data = (data << 4) >> 4;
```

Comment:
nitpick: data &= 0xFFF0

Reply:
This is for sign-extension of temp value. The << will cause it to lose the 4 MSB. In the >> it go back to original form but MSB is now sign extended.

Reply:
sorry: I ment data &= 0xFFF